core/finality-grandpa/src/environment.rs

```diff
@@ -801,6 +797,19 @@ where
 		round: RoundNumber,
 		commit: Commit<Block>,
 	) -> Result<(), Self::Error> {
+		let status = self.inner.info().chain;
```

Comment:
Why did you moved this up?

Reply:
there is also a significant change in which this doesn't access `client.backend()` anymore. however, you are correct all this can and should probably still be done internally.